version.js

```diff
@@ -1,6 +1,49 @@
 const github = require('@actions/github')
 const semver = require('semver')
 
+const DEFAULT_VERSION = '0.0.0'
+
+async function getCommitsOnBranch(branch, config) {
+    const commits = new Set()
+    // eslint-disable-next-line no-restricted-syntax
+    for await (const response of config.octokit.paginate.iterator(
+        config.octokit.rest.repos.listCommits,
+        {
+            ...github.context.repo,
+            sha: branch,
+        },
+    )) {
+        response.data.forEach((commit) => {
+            commits.add(commit.sha)
+        })
+    }
+    return commits
+}
+
+async function getLatestVersionInCommits(commits, sortedVersions, objectsByVersion, config) {
+    for (let i = 0; i < sortedVersions.length; i++) {
+        const refObj = objectsByVersion[sortedVersions[i]]
+
+        if (refObj.type === 'commit' && commits.has(refObj.sha)) {
+            return `${sortedVersions[i]}`
+        }
+
+        if (refObj.type === 'tag') {
+            // eslint-disable-next-line no-await-in-loop
```

Comment:
[eslint doesn't like `await` in a loop because it would prefer you to create all the promises at once, then get the resuls using Promise.all()](https://eslint.org/docs/rules/no-await-in-loop). However, in this case, the iterations of a loop are not actually independent of each-other because there is no need to consider other versions once the greatest version on the branch has been found.